compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -576,7 +576,7 @@ public void visit(BLangInvocation invocationExpr) {
 
             // Todo: test lambdas and function arguments
 
-            BSymbol dependsOnFunctionSym = symResolver.lookupSymbol(this.env, name, SymTag.FUNCTION);
+            BSymbol dependsOnFunctionSym = symResolver.lookupMainSpaceSymbol(this.env, name);
```

Comment:
We need to add SymTag validation here.

Reply:
Same comment as above. For phases after semantic analyzer, we don't need to validate the sym tag since it would already have been caught if the sym tag was invalid.